src/features/hotspots/root/HotspotsView.tsx

```diff
@@ -225,11 +234,28 @@ const HotspotsView = ({
     setDetailHeaderHeight(event.nativeEvent.layout.height)
   }, [])
 
+  const onMapHexSelected = useCallback(
+    async (hexId: string) => {
+      setSelectedHexId(hexId)
+      setSelectedHotspotIndex(0)
+      const result = (await dispatch(fetchHotspotsForHex({ hexId }))) as {
+        payload?: Hotspot[]
+      }
+      if (result && result.payload && result.payload.length) {
```

Comment:
wondering if you could use the safe accessor pattern for things like this?

`if (result?.payload?.length) {`

Reply:
Updated